doc/changelog.md

```diff
@@ -63,104 +108,96 @@
   array(1.0)
   ```
 
-* The AutoGraph feature, still experimental, now supports native Python `while` loops as well.
-  [(#318)](https://github.com/PennyLaneAI/catalyst/pull/318)
+  Automatic type promotion across conditional branches also now works with AutoGraph:
```

Comment:
Is this necessary for the changelog, seems like it is just an implicit effect of the above no?

Reply:
It is an implicit effect of the above, but wanted to make it clear (definitely feels like a 'free' additional improvement)

Reply:
```suggestion
  Automatic type promotion across conditional branches also works with AutoGraph:
```